### PR TITLE
auto map sage intacct employees

### PR DESCRIPTION
**Pull request by NileshPant1999**
charge
conffug

### CODE REVIEW

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.html [hunk 1]
```diff
@@ -59,6 +59,29 @@
         </mat-form-field>
       </div>
 
+      <div class="configurations--label">
+        Auto Map Employees from Fyle
+      </div>
+      <div fxLayout="row" fxLayoutAlign="start center">
+        <mat-form-field floatLabel="always" appearance="outline" fxFlex="100">
+          <mat-select placeholder="Choose how to auto map employees between Fyle and NetSuite"
```
Comment (Sravanksk) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.html:67]: `Sage Intacct`

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.html [hunk 2]
```diff
@@ -59,6 +59,29 @@
...
+            formControlName="autoMapEmployees">
+            <mat-option>
+              None
+            </mat-option>
+            <mat-option [value]="'EMAIL'">
+              Match emails on Fyle and Quickbooks online
```
Comment (Sravanksk) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.html:73]: `Sage Intacct`

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.html [hunk 3]
```diff
@@ -59,6 +59,29 @@
...
+            </mat-option>
+            <mat-option [value]="'NAME'">
+              Match names on Fyle and Quickbooks Online
+            </mat-option>
+            <mat-option [value]="'EMPLOYEE_CODE'">
+              Match Fyle Employee Code to Quickbooks Online Name
```
Comment (Sravanksk) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.html:79]: same

File: src/app/si/settings/employee-mappings/employee-mappings.component.html [hunk 4]
```diff
@@ -11,6 +11,9 @@
     <button class="page-info--primary-action" mat-flat-button color="primary" type="button" (click)="open()">
       Create Employee Mapping
     </button>
+    <button class="secondary-button" mat-flat-button color="secondary" type="button" (click)="triggerAutoMapEmployees()">
```
Comment (Sravanksk) [inline src/app/si/settings/employee-mappings/employee-mappings.component.html:14]: show this conditionally - 
`*ngIf="generalSettings. auto_map_employees"`

Comment (NileshPant1999) [reply src/app/si/settings/employee-mappings/employee-mappings.component.html:14]: okay

**Review by ashwin1111 — APPROVED** [review]

File: src/app/si/settings/category-mappings/category-mappings-dialog/category-mappings-dialog.component.ts [hunk 5]
```diff
@@ -82,7 +82,8 @@ export class CategoryMappingsDialogComponent implements OnInit {
           source_type: 'CATEGORY',
           destination_type: that.generalSettings.reimbursable_expenses_object === 'BILL' ? 'ACCOUNT' : 'EXPENSE_TYPE',
           source_value: fyleCategory.value,
-          destination_value: that.generalSettings.reimbursable_expenses_object === 'BILL' ? sageIntacctAccount.value : sageIntacctExpenseTypes.value
+          destination_value: that.generalSettings.reimbursable_expenses_object === 'BILL' ? sageIntacctAccount.value : sageIntacctExpenseTypes.value,
+          destination_id: that.generalSettings.reimbursable_expenses_object === 'BILL' ? sageIntacctAccount.destination_id : sageIntacctExpenseTypes.destination_id,
```
Comment (ashwin1111) [inline src/app/si/settings/category-mappings/category-mappings-dialog/category-mappings-dialog.component.ts:86]: rm `,`

File: src/app/si/settings/employee-mappings/employee-mappings-dialog/employee-mappings-dialog.component.ts [hunk 6]
```diff
@@ -58,29 +58,31 @@ export class EmployeeMappingsDialogComponent implements OnInit {
 
   submit() {
     const that = this;
-
     const fyleEmployee = that.form.controls.fyleEmployee.value;
     const sageIntacctVendor = that.generalSettings.reimbursable_expenses_object === 'BILL' ? that.form.value.sageIntacctVendor : '';
     const sageIntacctEmployee = that.generalSettings.reimbursable_expenses_object === 'EXPENSE_REPORT' ? that.form.value.sageIntacctEmployee : '';
     const creditCardAccount = that.form.value.creditCardAccount ? that.form.value.creditCardAccount.value : null;
+    const creditCardAccountId = that.form.value.creditCardAccount ? that.form.value.creditCardAccount.destination_id : that.generalMappings.default_charge_card_name;
```
Comment (ashwin1111) [inline src/app/si/settings/employee-mappings/employee-mappings-dialog/employee-mappings-dialog.component.ts:65]: why `that.generalMappings.default_charge_card_name` ?

File: src/app/si/settings/employee-mappings/employee-mappings.component.scss [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    max-width: 24px;
+    max-height: 24px;
+    font-size: 20px;
+    vertical-align: middle;
+}
+
+.secondary-button {
+    margin-left: 15px;
+    background-color: #dbdbdbde;
+}
```
Comment (ashwin1111) [inline src/app/si/settings/employee-mappings/employee-mappings.component.scss:15]: new line

File: src/app/core/models/mapping-destination.model.ts [hunk 8]
```diff
@@ -8,5 +8,7 @@ export class MappingDestination {
   destination_id: number;
   created_at: Date;
   updated_at: Date;
-  workspace: number
-}
\ No newline at end of file
+  workspace: number;
+  detail: any;
```
Comment (Shwetabhk) [inline src/app/core/models/mapping-destination.model.ts:12]: we'll know the structure of this right? can't we add this
```js
details: {
    email: string;
};
```

File: src/app/si/settings/category-mappings/category-mappings-dialog/category-mappings-dialog.component.ts (hunk 5 above)
Comment (Shwetabhk) [reply src/app/si/settings/category-mappings/category-mappings-dialog/category-mappings-dialog.component.ts:86]: +1

File: src/app/core/models/mapping-destination.model.ts (hunk 8 above)
Comment (ashwin1111) [reply src/app/core/models/mapping-destination.model.ts:12]: hey destination can be of any structure. We have different structure of different attributes.
Ex:
```
details: {
    CUSTOMERID: string,
    CUSTOMERNAME: string
}
```

File: src/app/si/settings/employee-mappings/employee-mappings-dialog/employee-mappings-dialog.component.ts (hunk 6 above)
Comment (NileshPant1999) [reply src/app/si/settings/employee-mappings/employee-mappings-dialog/employee-mappings-dialog.component.ts:65]: changing it to that.generalMappings.default_charge_card_id

File: src/app/core/models/mapping-destination.model.ts (hunk 8 above)
Comment (NileshPant1999) [reply src/app/core/models/mapping-destination.model.ts:12]: yes we can do this

File: src/app/si/settings/employee-mappings/employee-mappings-dialog/employee-mappings-dialog.component.ts (hunk 6 above)
Comment (ashwin1111) [reply src/app/si/settings/employee-mappings/employee-mappings-dialog/employee-mappings-dialog.component.ts]: even that is not right

**Pull request merged**